x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/sequence/SequenceMatcher.java

```diff
@@ -339,4 +340,27 @@ public String toString() {
                 completed.size(),
                 stageToKeys);
     }
+
+    private static long timestampDiffNanos(Object ts1, Object ts2) {
+        long timestampDiffNanos;
+        if (ts1 instanceof Instant) {
+            Instant i1 = (Instant) ts1;
+            Instant diff;
+            if (ts2 instanceof Instant) {
+                Instant i2 = (Instant) ts2;
+                diff = i1.minusNanos(i2.getNano()).minusSeconds(i2.getEpochSecond());
+            } else {
+                diff = i1.minusMillis(((Number) ts2).longValue());
+            }
+            timestampDiffNanos = diff.getEpochSecond() * 1_000_000_000L + diff.getNano();
```

Comment:
```suggestion
            return diff.getEpochSecond() * 1_000_000_000L + diff.getNano();
```

and the same for the other branches.

Reply:
Refactored.